File: iosApp/PrendsEtRends/Models/LoanItem.swift
import Foundation

public struct LoanItem: Identifiable, Codable, Equatable {
    public var id: Int64
    public var title: String
    public var type: LoanType
    public var contactName: String
    public var contactPhone: String?
    public var contactEmail: String?
    public var valueOrCategory: String?
    public var photoPath: String?
    public var audioPath: String?
    public var audioDurationMs: Int64
    public var mediaList: [MediaItem]
    public var dueDate: Int64? // Epoch in milliseconds
    public var createdDate: Int64
    public var isReturned: BooleanLiteralType
    public var returnedDate: Int64?
    public var notes: String?

    public init(
        id: Int64 = Int64(Date().timeIntervalSince1970 * 1000) + Int64.random(in: 0...999),
        title: String,
        type: LoanType,
        contactName: String,
        contactPhone: String? = nil,
        contactEmail: String? = nil,
        valueOrCategory: String? = nil,
        photoPath: String? = nil,
        audioPath: String? = nil,
        audioDurationMs: Int64 = 0,
        mediaList: [MediaItem] = [],
        dueDate: Int64? = nil,
        createdDate: Int64 = Int64(Date().timeIntervalSince1970 * 1000),
        isReturned: Bool = false,
        returnedDate: Int64? = nil,
        notes: String? = nil
    ) {
        self.id = id
        self.title = title
        self.type = type
        self.contactName = contactName
        self.contactPhone = contactPhone
        self.contactEmail = contactEmail
        self.valueOrCategory = valueOrCategory
        self.photoPath = photoPath
        self.audioPath = audioPath
        self.audioDurationMs = audioDurationMs
        self.mediaList = mediaList
        self.dueDate = dueDate
        self.createdDate = createdDate
        self.isReturned = isReturned
        self.returnedDate = returnedDate
        self.notes = notes
    }

    public var isOverdue: Bool {
        guard !isReturned, let due = dueDate else { return false }
        return due < Int64(Date().timeIntervalSince1970 * 1000)
    }
}

File: app/src/main/java/com/example/data/LoanItem.kt
package com.example.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "loans")
data class LoanItem(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val type: LoanType, // LENT or BORROWED
    val contactName: String,
    val contactPhone: String? = null,
    val contactEmail: String? = null,
    val valueOrCategory: String? = null, // e.g. "50 €" or "Objet / Outil"
    val photoPath: String? = null,
    val audioPath: String? = null,
    val audioDurationMs: Long = 0,
    val mediaList: List<com.example.media.MediaItem> = emptyList(),
    val dueDate: Long? = null, // Epoch timestamp in ms
    val createdDate: Long = System.currentTimeMillis(),
    val isReturned: Boolean = false,
    val returnedDate: Long? = null,
    val notes: String? = null
) {
    val isOverdue: Boolean
        get() = !isReturned && dueDate != null && dueDate < System.currentTimeMillis()
}
